core/chains/evm/client/evm_client.go

```diff
@@ -20,7 +20,7 @@ func NewEvmClient(cfg evmconfig.NodePool, chainCfg commonclient.ChainConfig, cli
 	var sendonlys []commonclient.SendOnlyNode[*big.Int, RPCClient]
 	largePayloadRPCTimeout, defaultRPCTimeout := getRPCTimeouts(chainType)
 	for i, node := range nodes {
-		if node.SendOnly != nil && *node.SendOnly {
+		if node.WSURL == nil || (node.SendOnly != nil && *node.SendOnly) {
```

Comment:
A secondary or SendOnly node has limited functionality. It's only used to broadcast transactions (fire and forget), and we are not running health checks for it.
We should not change the type of the node. Instead, introduce changes to the RPC client.

Reply:
Moving the check to `Dial()` at rpc_client.go